### ISSUE TITLE
GPU memory increases during trainning

### ISSUE THREAD

**Issue by KleinXin** [NVIDIA/DALI#2943]
I am using tfrecord to load data, the GPU memory increases from about 11G to 15G during trainning in one epoch. So I have to decrease the batch_size in case OOM error occurs.
However, when default pytorch dataloader is used, the GPU memory does not increase dramatically.

The codes that used to get input and lables are shown below

trainingPipes = TFRecordPipeline(tfrecord_path=train_tfrecord_path,
                                     tfrecord_idx_path=train_tfrecord_idx,
                                     batch_size=batch_size, 
                                     num_threads=4, 
                                     device_id = 0,
                                     crop_size = crop_size,
                                     image_size = img_size,
                                     randomFlip = True,
                                     is_color_jitter = False,
                                     randomTranspose = random_transpose,
                                     random_transpose_thresh=random_transpose_thresh,
                                     random_shuffle=True)

        trainingPipes.build()

        trainloader = DALIClassificationIterator(
                      trainingPipes,
                      size=int(trainingPipes.epoch_size("Reader"))
        ) 

        for i, data in enumerate(trainloader):

            images = data[0]["data"].cuda()

            labels_ = data[0]["label"].cpu().numpy()

            labels = []
            for label in labels_:
                labels.append(label[0])

            labels = torch.FloatTensor(labels).long().cuda()

numpy array is used to rearrange the lables because data[0]["label"] is a list of list. It is in the format [[label_1],[label_2],[label_3]...]

**Comment by JanuszL** [comment]
Hi @KleinXin,

That is expected that when you use DALI with GPU operators it consumes more GPU memory than the PyTorch data loader which runs purely on CPU.
You can read more about memory consumption in [this documentation section](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/advanced_topics_performance_tuning.html#memory-consumption).

**Comment by KleinXin** [comment]
> Hi @KleinXin,
> 
> That is expected that when you use DALI with GPU operators it consumes more GPU memory than the PyTorch data loader which runs purely on CPU.
> You can read more about memory consumption in [this documentation section](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/advanced_topics_performance_tuning.html#memory-consumption).

I am afraid some temporary variables during processing is not released after using. But I do not know where I can change.
